### PR TITLE
Add OP Granite fork

### PR DESCRIPTION
**Pull request by benaadams**
## Changes

- Add OP Granite fork

#### What types of changes does your code introduce?

- [x] New feature (a non-breaking change that adds functionality)

#### Requires testing

- [ ] Yes
- [ ] No

#### If yes, did you write tests?

- [ ] Yes
- [ ] No

#### Notes on testing

_Optional. Remove if not applicable._

#### Requires documentation update

- [ ] Yes
- [ ] No

_If yes, link the PR to the docs update or the issue with the details labeled `docs`. Remove if not applicable._

#### Requires explanation in Release Notes

- [ ] Yes
- [ ] No

_If yes, fill in the details here. Remove if not applicable._

## Remarks

_Optional. Remove if not applicable._

### CODE REVIEW

File: src/Nethermind/Nethermind.Network/ProtocolValidator.cs [hunk 1]
```diff
@@ -68,7 +68,7 @@ private bool ValidateEthProtocol(ISession session, ProtocolInitializedEventArgs
 
             if (_forkInfo.ValidateForkId(syncPeerArgs.ForkId.Value, _blockTree.Head?.Header) != ValidationResult.Valid)
             {
-                return Disconnect(session, DisconnectReason.InvalidForkId, CompatibilityValidationType.InvalidForkId, "invalid fork id");
+                return Disconnect(session, DisconnectReason.InvalidForkId, CompatibilityValidationType.InvalidForkId, $"invalid network id {syncPeerArgs.NetworkId} fork id {syncPeerArgs.ForkId.Value}");
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Network/ProtocolValidator.cs:71]: would be good to add `ValidationResult` to the message

File: src/Nethermind/Chains/op-sepolia.json [hunk 2]
```diff
@@ -62,6 +63,7 @@
     "eip5656TransitionTimestamp": "0x65D62C10",
     "eip6780TransitionTimestamp": "0x65D62C10",
     "rip7212TransitionTimestamp": "0x66575100",
+    "OpGraniteTransitionTimestamp": "0x66ba3180",
```
Comment (flcl42) [inline src/Nethermind/Chains/op-sepolia.json:66]: could you make it `opGraniteTransitionTimestamp`

**Review by flcl42 — APPROVED** [review]

**Review by LukaszRozmej — APPROVED** [review]

File: src/Nethermind/Chains/op-sepolia.json (hunk 2 above)
Comment (benaadams) [reply src/Nethermind/Chains/op-sepolia.json:66]: Done

File: src/Nethermind/Nethermind.Network/ProtocolValidator.cs (hunk 1 above)
Comment (benaadams) [reply src/Nethermind/Nethermind.Network/ProtocolValidator.cs]: Done

**Pull request merged**